firmware/sys/system_init/system_init.c

```diff
@@ -0,0 +1,149 @@
+#include "stdio.h"
+#include "ds18_sensor.h"
+#include "moisture_sensor.h"
+#include "xtimer.h"
+#include "serialization.h"
+#include "udp_m4a.h"
+#include "board.h"
+#include "thread.h"
+
+#include "net/gnrc/netif.h"
+#include "net/netif.h"
+
+#ifdef CPU_ESP32
+
+#include "esp_system.h"
+#define SLIP_IFACE (10)
+
+#endif
+
+#ifdef DS18B20_PIN
+#define DS18_PIN  DS18B20_PIN
+#else 
+#define DS18_PIN  0  /* is recommended add DS18B20 PIN since the settings of your board,
+                         this variable is for avoid compilations error in other boards. */
+#endif
+
+#define SERVER_BUFFER_SIZE      (100)
+
+int ds18_is_avaliable = 1;
+int moisture_sensor_is_avaliable = 1;
+uint8_t buffer[SERVER_BUFFER_SIZE];
+
+
+char get_sensor_event[THREAD_STACKSIZE_DEFAULT];
+
+void *init_loop (void *args) {
+    int port = 3000;
+    char address[30] = "::1";
+    sensor_data payload; 
+    while(1){
+    
+        if(moisture_sensor_is_avaliable != 0)
+        {
+            if(get_moisture_value(&payload.soil_moisture) < 0)
+            {
+                payload.soil_moisture = 0;
+                printf("Error to the obtain the moisture \n");
+            }
+        }
+        if(ds18_is_avaliable != 0)
+        {
+            if(get_temperature_value(&payload.temperature) < 0) 
+            {
+                payload.temperature = 0;
+                printf("Error to the obtain the temperature \n"); 
+            }
+        }
+
+        if(ds18_is_avaliable != 0 || moisture_sensor_is_avaliable != 0)
+        {   
+             printf("the soil_moisture is %i and temperature is %i \n", payload.soil_moisture, payload.temperature);
+             size_t len_cbor = sizeof(sensor_data);
+             if(cbor_enconde_message(&payload, buffer, &len_cbor) < 0) {
+                 printf("error to encode payload with cbor");
+             } else {
+                udp_send(&port, address, buffer, &len_cbor);
+             }
+        }   
+        xtimer_sleep(60);
+    }
+}
+
+
+int set_wired_ipv6 (void)
+{   
+    #ifdef ESP32
+    uint8_t mac[6];
+    if(esp_base_mac_addr_get(mac) < 0){
+        printf("error with the mac here!!!!!!!!!!!!! \n");
```

Comment:
:slightly_frowning_face:

Reply:
why do you sad? i think i must fix the printf, but i not totally sure xD